packages/react-internal/src/components/Modal/Modal.base.tsx

```diff
@@ -209,29 +146,245 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
+    const allowScrollOnModal = React.useCallback(
+      (elt: HTMLDivElement | null): void => {
+        if (elt) {
+          if (internalState.allowTouchBodyScroll) {
+            allowOverscrollOnElement(elt, internalState.events);
+          } else {
+            allowScrollOnElement(elt, internalState.events);
+          }
+        } else {
+          internalState.events.off(internalState.scrollableContent);
+        }
+        internalState.scrollableContent = elt;
+      },
+      [internalState],
+    );
+
+    const registerInitialModalPosition = (): void => {
+      if (dragOptions?.keepInBounds && !internalState.minClampedPosition && !internalState.maxClampedPosition) {
+        const dialogMain = doc?.querySelector(`[data-id=${FocusTrapZoneId}]`);
+        if (dialogMain) {
+          const modalRectangle = dialogMain.getBoundingClientRect();
+          internalState.minClampedPosition = { x: -modalRectangle.x, y: -modalRectangle.y };
+          internalState.maxClampedPosition = { x: modalRectangle.x, y: modalRectangle.y };
+        }
+      }
+    };
+
+    /**
+     * Clamps an axis to a specified min and max position.
+     *
+     * @param axis A string that represents the axis (x/y).
+     * @param position The position on the axis.
+     */
+    const getClampedAxis = React.useCallback(
+      (axis: string, position: number) => {
+        if (!dragOptions || !dragOptions.keepInBounds) {
+          return position;
+        }
+
+        position = Math.max(internalState.minClampedPosition[axis as keyof ICoordinates], position);
+        position = Math.min(internalState.minClampedPosition[axis as keyof ICoordinates], position);
+
+        return position;
+      },
+      [dragOptions, internalState],
+    );
+
+    const handleKeyUp = React.useCallback(
+      (ev: React.KeyboardEvent<HTMLElement>): void => {
+        // Need to handle the CTRL + ALT + SPACE key during keyup due to FireFox bug:
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1220143
+        // Otherwise it would continue to fire a click even if the event was cancelled
+        // during mouseDown.
+        if (ev.altKey && ev.ctrlKey && ev.keyCode === KeyCodes.space) {
+          // Since this is a global handler, we should make sure the target is within the dialog
+          // before opening the dropdown
+          if (elementContains(internalState.scrollableContent, ev.target as HTMLElement)) {
+            toggleModalMenuOpen();
+            ev.preventDefault();
+            ev.stopPropagation();
+          }
+        }
+      },
+      [internalState, toggleModalMenuOpen],
+    );
+
+    const handleModalClose = React.useCallback((): void => {
+      internalState.lastSetXCoordinate = 0;
+      internalState.lastSetYCoordinate = 0;
+
+      setXCoordinate(0);
+      setYCoordinate(0);
+      setModalMenuClose();
+      setIsModalOpen(false);
+      setKeyboardMoveModeFalse();
+
+      if (dragOptions && internalState.hasRegisteredKeyUp) {
```

Comment:
For cases like this where you only care about the *presence* of dragOptions (and not any of its current values), you could optionally get rid of the dep on the whole object (which might mutate frequently) by doing `const hasDragOptions = !!dragOptions` at the outer scope and then using that as the dependency in the callback. Not sure how important that is though.